core/modules/admin_bar/admin_bar.inc

```diff
@@ -816,6 +874,11 @@ function admin_bar_theme_settings() {
   $form['components']['admin_bar.page']['#description'] = t('Links to edit items that make up the current page (e.g. layout, theme).');
   $form['components']['admin_bar.users']['#description'] = t('A count of the people currently logged into the site.');
   $form['components']['admin_bar.account']['#description'] = t('Links to view and log out of the current user account.');
+  $form['components']['admin_bar.locale']['#description'] = t('Links to switch the language of the user interface. This option is available only when the Locale module is enabled.');
+
+  if (module_exists('locale')) {
+    $form['components']['admin_bar.locale']['#disabled'] = FALSE;
+  }
```

Comment:
These lines here wouldn't be required if we changed the `'#disabled'` property like I proposed in my previous comment.